### DISCUSSION TITLE
Best way to find records with an attachment?

### DISCUSSION THREAD

**Discussion by dan** [shrinerb/shrine#659] [General]
I have Car model which can have an attached Photo. Using ActiveRecord, I'd like to find just the Cars with a Photo attached. I've been checking for a null `photo_data` column, but I'm wondering if this is the best way, or if there's built-in method I should be using to do this. I can't seem to find one.

Currently it looks like this:

```
Car.where(photo_data: nil)
```

And the scope:

```
scope :photographed, -> { where.not(photo_data: nil) }
```

Thanks for any tips!

**Comment by benkoshy** [comment]
I do not know of a better way, nor can I think of a better way, than what you've done above.
